### ISSUE TITLE
ttf() output becomes NaN if input increases

### ISSUE THREAD

**Issue by belm0** [VictoriaMetrics/VictoriaMetrics#757]
**Describe the bug**
The output of `ttf(vm_free_disk_space_bytes{job="victoria-metrics"})` is usually NaN.

I tried it as a range query and instant query.

The function is not documented in detail, so I don't know how many data points are considered, etc. to judge the output.

**Screenshots**
![Screen Shot 2020-09-09 at 12 40 34 PM](https://user-images.githubusercontent.com/1708631/92551854-1b973080-f29a-11ea-814d-caa46f9528b5.png)
![Screen Shot 2020-09-09 at 12 43 20 PM](https://user-images.githubusercontent.com/1708631/92551859-1f2ab780-f29a-11ea-9ec0-4157e76510f5.png)

**Version**
v1.40.0

**Comment by belm0** [comment]
Here is an attempt with `[2d]` vector.  The series abruptly stops.

I can speculate that perhaps `ttf()` doesn't handle increasing free space, which may happen due to compression etc.

![Screen Shot 2020-09-09 at 12 47 55 PM](https://user-images.githubusercontent.com/1708631/92552117-dfb09b00-f29a-11ea-8b7a-3230301453b1.png)

**Comment by belm0** [comment]
if `ttf()` doesn't produce a usable value at the end of the range, it cannot be used with instant query for alerting, etc.

**Comment by belm0** [comment]
evidence that NaN output may be related to NaN of input

query here is `ttf(vm_free_disk_space_bytes{job="victoria-metrics"}[1h])`

![Screen Shot 2020-09-09 at 1 49 56 PM](https://user-images.githubusercontent.com/1708631/92555703-37530480-f2a3-11ea-8558-8623fe418ee2.png)

**Comment by belm0** [comment]
conclusion:  it's caused by increases in source series

![Screen Shot 2020-09-09 at 2 02 54 PM](https://user-images.githubusercontent.com/1708631/92556567-3cb14e80-f2a5-11ea-8f5f-a69c69eaff92.png)

**Comment by belm0** [comment]
aside:  for the original use case of estimating VM storage full, I found it's much better to derive it from ingestion rate rather than free storage rate

```
# seconds_until_full_estimate = free_bytes / (datapoint_rate[1d] * bytes_per_datapoint)
vm_free_disk_space_bytes / (sum(rate(vm_rows_inserted_total[1d])) *
    (sum(vm_data_size_bytes{type !="indexdb"}) / sum(vm_rows{type != "indexdb"})))
```

**Comment by valyala** [comment]
> for the original use case of estimating VM storage full, I found it's much better to derive it from ingestion rate rather than free storage rate

This looks much better than generic `ttf(vm_free_disk_space_bytes)`! Thanks for doing this!

As for missing results for `ttf()`, could you build VictoriaMetrics from commit 2114179e1935e1e1ab3376638419b69a62ac37e7  and verify whether it fixes the issue with missing `ttf()` results? See [build instructions](https://victoriametrics.github.io/#how-to-build-from-sources)? It is also recommended passing big time range inside square brackets to `ttf` in order to get smooth graphs. For example, `ttf(m[1h])`.

**Comment by valyala** [comment]
FYI, the commit mentioned above has been included in [v1.41.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by belm0** [comment]
ttf() function appears to be fixed
